File: app/src/main/java/com/yuk/miuiHomeR/Application.java
package com.yuk.miuiHomeR;

import android.content.Context;

import com.yuk.miuiHomeR.utils.PrefsUtils;

public class Application extends android.app.Application {

    @Override
    protected void attachBaseContext(Context base) {
        PrefsUtils.mSharedPreferences = PrefsUtils.getSharedPrefs(base, false);
        super.attachBaseContext(base);
    }
}

File: app/src/main/java/com/yuk/miuiHomeR/Application.kt
package com.yuk.miuiHomeR

import android.content.Context
import com.yuk.miuiHomeR.utils.PrefsUtils
import com.yuk.miuiHomeR.utils.PrefsUtils.getSharedPrefs
import com.yuk.miuiHomeR.utils.ktx.setLocale
import java.util.Locale

class Application : android.app.Application() {

    override fun attachBaseContext(base: Context?) {
        PrefsUtils.mSharedPreferences = base?.let { getSharedPrefs(it, false) }
        val locale: String? = PrefsUtils.mSharedPreferences?.getString("prefs_key_settings_language", "SYSTEM")
        if (locale != null) {
            if (base != null)
                super.attachBaseContext(setLocale(base, Locale.forLanguageTag(locale)))
        } else super.attachBaseContext(base)
    }
}